### DISCUSSION TITLE
模板编译不了

### DISCUSSION THREAD

**Discussion by checkpoint-x** [sikouhjw/gdutthesis#45] [Q&A] [answered]
最新的模板下载后无法编译，并且有以下报错
![image](https://user-images.githubusercontent.com/101620684/158314954-0cd07b32-5923-4d98-b529-3784582bbf94.png)

请问应该如何解决，谢谢

**Comment by sikouhjw** [comment, marked as answer]
1. 除非必要，否则请从 `README.md` 里的 Overleaf 链接使用模板；
2. 如果要手动上传模板，请将编译方式改为 `xelatex`；
3. Overleaf 如何改编译方式？左侧菜单栏。

**Reply by checkpoint-x** [reply]
好的，谢谢，问题已解决。还有个问题，表格里的字体如何设置成比正文小一号？

**Reply by sikouhjw** [reply]
> 还有个问题，表格里的字体如何设置成比正文小一号？

现在没有小一号吗？

**Reply by checkpoint-x** [reply]
嗷嗷，新的模板有了，我的问题，谢谢你

**Reply by checkpoint-x** [reply]
![8f32c074383c93e751a9b1a3c2b7f70](https://user-images.githubusercontent.com/101620684/158336639-4f329f95-2726-45e8-9b7e-f989fca9d2f9.png)
![dbf8db22797bdfdb9b9ea7834d6ce5e](https://user-images.githubusercontent.com/101620684/158336654-66f6184a-d052-475c-b5c7-656bab4933c1.png)
我是按着模板把信息往上面填的，想问一下这种情况应该怎么调。

**Reply by sikouhjw** [reply]
> 我是按着模板把信息往上面填的，想问一下这种情况应该怎么调。

https://github.com/sikouhjw/gdutthesis/issues/19

**Reply by checkpoint-x** [reply]
往导言区加代码和替换cls文件后，还是不行

**Reply by sikouhjw** [reply]
> 往导言区加代码和替换cls文件后，还是不行

Overleaf 链接分享一下，改完了再关掉分享权限。

**Reply by checkpoint-x** [reply]
https://cn.overleaf.com/4969479951tvzwbtsmhxpn
麻烦你了

**Reply by sikouhjw** [reply]
> https://cn.overleaf.com/4969479951tvzwbtsmhxpn 麻烦你了

1. 357 行的 `\gdutcite*{xiaoqiang2014stability}` 是需要用最新的 cls 才能用的，而你的 cls 没有更新，所以会报错；
2. `\cite` 不能放段首，所以要在前面加一点文字；
3. 将那段代码放进导言区后是没有问题的；
4. 论文太大了，Overleaf 编译时间是有限的，所以我在弄的时候很容易编译不成功（也可能是网络问题？）。

**Reply by checkpoint-x** [reply]
谢谢大佬 谢谢大佬
更新是把那个压缩包下下来然后解压，再上传那个cls文件，把原来的替换么？

**Reply by sikouhjw** [reply]
> 谢谢大佬 谢谢大佬 更新是把那个压缩包下下来然后解压，再上传那个cls文件，把原来的替换么？

是的，把 cls 换一下就可以了，其它的没变什么。可以关闭分享权限了。

**Reply by checkpoint-x** [reply]
但其实我前面就是这么做的，下载的是#19 里面的压缩包
再次谢谢大佬

**Reply by sikouhjw** [reply]
> 但其实我前面就是这么做的，下载的是#19 里面的压缩包 再次谢谢大佬

逻辑是这样的，推荐使用『发行版』而不是紧跟『最新版』，所以你下载的是『最后一个发行版』，而 `\gdutcite` 是之后才有的功能，所以会报错，所以你需要去下载的是这个 [gdutthesis.cls](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls)，然后再传上去。
